lib/internal/modules/esm/formats.js

```diff
@@ -23,6 +24,10 @@ if (experimentalWasmModules) {
   extensionFormatMap['.wasm'] = 'wasm';
 }
 
+if (experimentalAddonModules) {
+  extensionFormatMap['.node'] = 'addon';
```

Comment:
I am somewhat surprised to find that there is not a test for the `node-addons` exports condition...it seems the the files pointed to by `node-addons` won't get interpreted as addons unless they have the `.node` extension, though this issue predates this PR and also happens to `require()`.

Reply:
I think `node-addons` condition is not changed in this PR. Do you mind if I track it in a follow up?

Taking a step back, I'm wondering if the format name could be renamed as `node-addon`/`node-addons` for alignment.

Reply:
Sure, also I am not even sure if anyone is actually using it, it seems only useful to distinguish "if the current run times can load Node.js style addons" as it currently stands, so may not really be that big of a deal anyway....

Reply:
I prefer the more concise `addon`

Reply:
Actually I think I misread https://github.com/nodejs/node/pull/55844#discussion_r1871753939 - I think for the format name, `addon` or `node-addon` would make more sense. `node-addons` would be a bit weird in the hooks since it's plural.

Reply:
This format enum is specific for Node.js so I'll stick with the concise `addon`.

Reply:
In any case, the exports condition question is separate (it's more about when a module that doesn't end with `.node` is pointed to by `node-addons`, should it be interpreted(translated) into an addon or should it be interpreted based on whatever extension it has - and if it has a different extension (e.g. dylib or dll), what should happen then).

Reply:
The reality is that the conditional exported entries are still been loaded based on their extensions, not their entry types. With `require(esm)`, it is possible to do the following already :eyes::

```
{
  "exports": {
    "import": "./index-module.cjs",
    "require": "./index-require.mjs"
  }
}
```